### PR TITLE
Add Backlog to SocketTransportOptions

### PR DESCRIPTION
**Pull request by Kahbazi**
Addresses #13270

### CODE REVIEW

**Comment by Tratcher** [comment]
Adding APIs requires updating the reference assemblies.
https://github.com/aspnet/AspNetCore/blob/master/docs/ReferenceAssemblies.md

**Comment by jkotalik** [comment]
Maybe add a unit test for sanity sake. Otherwise LGTM

**Review by jkotalik — APPROVED** [review]

**Review by davidfowl — COMMENTED** [review]
We should do the same for libuv

**Comment by Tratcher** [comment]
```
error : Generated code is not up to date in src/Servers/Kestrel/Transport.Sockets/ref/Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.netcoreapp.cs. You might need to regenerate the reference assemblies or project list (see docs/ReferenceAssemblies.md and docs/ReferenceResolution.md) [F:\workspace\_work\1\s\src\Servers\Kestrel\Transport.Sockets\ref\Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.netcoreapp.cs]
diff --git a/src/Servers/Kestrel/Transport.Sockets/ref/Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.netcoreapp.cs b/src/Servers/Kestrel/Transport.Sockets/ref/Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.netcoreapp.cs
index ee86fb1ad5..e54b0a2d21 100644
--- a/src/Servers/Kestrel/Transport.Sockets/ref/Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.netcoreapp.cs
+++ b/src/Servers/Kestrel/Transport.Sockets/ref/Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.netcoreapp.cs
@@ -19,8 +19,8 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets
     public partial class SocketTransportOptions
     {
         public SocketTransportOptions() { }
-        public int IOQueueCount { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute]set { } }
         public int Backlog { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute]set { } }
+        public int IOQueueCount { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute]set { } }
         public long? MaxReadBufferSize { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute]set { } }
         public long? MaxWriteBufferSize { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute]set { } }
         public bool NoDelay { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute]set { } }

Summary:

   1 error(s)

error : Generated code is not up to date in src/Servers/Kestrel/Transport.Sockets/ref/Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.netcoreapp.cs. You might need to regenerate the reference assemblies or project list (see docs/ReferenceAssemblies.md and docs/ReferenceResolution.md) [F:\workspace\_work\1\s\src\Servers\Kestrel\Transport.Sockets\ref\Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.netcoreapp.cs]
```
The reference assembly check is still failing. Did you update the ref assembly manually or using the tools I linked?

**Comment by Kahbazi** [comment]
@Tratcher I have problem with build so I did it manually 😟. I guess the properties should be sorted by alphabetical order, right?

**Comment by Tratcher** [comment]
Yeah. What was your problem with the build?

**Comment by Kahbazi** [comment]
This is the the logs when I run `build.cmd` for Kestrel solution
[build.txt](https://github.com/aspnet/AspNetCore/files/3741232/build.txt)

**Comment by Kahbazi** [comment]
@jkotalik what kind of test do you have in mind? All I can think of is a functional test which backlog is set to one, and when it receives two request, it expect an exception

**Comment by Tratcher** [comment]
> @jkotalik what kind of test do you have in mind? All I can think of is a functional test which backlog is set to one, and when it receives two request, it expect an exception

Yeah, there's not much to test here, we pass the option directly through. Don't worry about it.

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=392188

**Comment by Tratcher** [comment]
@davidfowl ready for API review.

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/3128
https://github.com/aspnet/AspNetCore-Internal/issues/2996

**Comment by halter73** [comment]
Do we want to review the API before the libuv changes are in?

**Comment by Tratcher** [comment]
This change is adequate in isolation. Libuv can follow in a separate PR. The public API is not complex.

**Comment by halter73** [comment]
Definitely not complex. So simple that I think we should merge both changes together so we don't risk forgetting about it.

**Comment by Kahbazi** [comment]
I will take a look at libuv and add Backlog to it.

**Comment by halter73** [comment]
Thanks @Kahbazi! The libuv backlog constant is currently used [here.](https://github.com/aspnet/AspNetCore/blob/a4d9faefe2ed6fbebd500413032336607ec92778/src/Servers/Kestrel/Transport.Libuv/src/Internal/ListenerPrimary.cs#L72-L73) and you can read the options from `TransportContext.Options` inside the ListenerPrimary class.

**Comment by Kahbazi** [comment]
I should also change [Listener](https://github.com/aspnet/AspNetCore/blob/a4d9faefe2ed6fbebd500413032336607ec92778/src/Servers/Kestrel/Transport.Libuv/src/Internal/Listener.cs#L40), right?

**Comment by halter73** [comment]
Correct. Good catch.

**Review by Tratcher — APPROVED** [review]

File: src/Servers/Kestrel/Transport.Libuv/ref/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.netcoreapp.cs [hunk 1]
```diff
@@ -14,7 +14,8 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv
     public partial class LibuvTransportOptions
     {
         public LibuvTransportOptions() { }
-        public long? MaxReadBufferSize { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute]set { } }
+        public int Backlog { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute]set { } }
+		public long? MaxReadBufferSize { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute]set { } }
```
[the remark this answers is not in the document]
Comment (Kahbazi) [reply src/Servers/Kestrel/Transport.Libuv/ref/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.netcoreapp.cs]: I did it manually, it wasn't showing on notepad!

[the remark this answers is not in the document]
Comment (Tratcher) [reply src/Servers/Kestrel/Transport.Libuv/ref/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.netcoreapp.cs]: Tabs vs spaces 😁

**Comment by Tratcher** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=393740

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/3241

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**